python/lsst/sims/catUtils/mixins/snObject.py

```diff
@@ -0,0 +1,718 @@
+            is the same for all elements of the dict.
+
+            if provided, overrides wavelen input and the SED is
+            obtained at the wavelength values native to bandpass
+            object.
+
+
+        Returns
+        -------
+        `sims_photutils.sed` object containing the wavelengths and SED
+        values from the SN at time time in units of ergs/cm^2/sec/nm
+
+
+        .. note: If both wavelen and bandpassobject are `None` then exception,
+                 will be raised.
+        Examples
+        --------
+        >>> sed = SN.SNObjectSED(time=0., wavelen=wavenm)
+        '''
+
+        if wavelen is None and bandpass is None:
+            raise ValueError('A non None input to either wavelen or\
+                              bandpassobject must be provided')
+
+        # if bandpassobject present, it overrides wavelen
+        if bandpass is not None:
+            if isinstance(bandpass, BandpassDict):
+                firstfilter = bandpass.keys()[0]
+                bp = bandpass[firstfilter]
+            else:
+                bp = bandpass
+            # remember this is in nm
+            wavelen = bp.wavelen
+
+        flambda = np.zeros(len(wavelen))
+
+        # self.mintime() and self.maxtime() are properties describing
+        # the ranges of SNCosmo.Model in time.
+
+        # Set SED to 0 beyond the model phase range, will change this if
+        # SNCosmo includes a more sensible decay later.
+        if (time > self.mintime()) & (time < self.maxtime()):
+
+            # If SNCosmo is requested a SED value beyond the model range
+            # it will crash. Try to prevent that by returning np.nan for
+            # such wavelengths. This will still not help band flux calculations
+            # but helps us get past this stage.
+
+            flambda = flambda * np.nan
+
+            # Convert to Ang
+            wave = wavelen * 10.0
+            mask1 = wave > self.minwave()
+            mask2 = wave < self.maxwave()
+            mask = mask1 & mask2
+            wave = wave[mask]
+
+            # flux density dE/dlambda returned from SNCosmo in
+            # ergs/cm^2/sec/Ang, convert to ergs/cm^2/sec/nm
+
+            flambda[mask] = self.flux(time=time, wave=wave)
+            flambda[mask] = flambda[mask] * 10.0
+
+        SEDfromSNcosmo = Sed(wavelen=wavelen, flambda=flambda)
+
+        if not applyExtinction:
+            return SEDfromSNcosmo
+
+        # Apply LSST extinction
+        ax, bx = SEDfromSNcosmo.setupCCMab()
+
+        if self.ebvofMW is None:
+            raise ValueError('ebvofMW attribute cannot be None Type and must'
+                             ' be set by hand using set_MWebv before this'
+                             'stage, or by using setcoords followed by'
+                             'mwEBVfromMaps\n')
+
+        SEDfromSNcosmo.addCCMDust(a_x=ax, b_x=bx, ebv=self.ebvofMW)
+        return SEDfromSNcosmo
+
+    def catsimBandFluxes(self, time, bandpassobject):
```

Comment:
should this method be named 'catsimBandFlux', since it only calculates flux in one bandpass?